dangerzone/isolation_provider/container.py

```diff
@@ -274,6 +275,15 @@ def _convert_with_tmpdirs(
             f"ENABLE_TIMEOUTS={self.enable_timeouts}",
         ]
         ret = self.exec_container(document, command, extra_args, stdout_callback)
+
+        if getattr(sys, "dangerzone_dev", False):
+            log_path = pixel_dir / "captured_output.txt"
+            with open(log_path, "r", encoding="ascii", errors="replace") as f:
+                untrusted_log = f.read(MAX_CONVERSION_LOG_CHARS)
+                document.conversion_output = self.sanitize_conversion_str(untrusted_log)
+            text = f"Container output:\n{document.conversion_output}"
+            self.print_progress(document, False, text, 50.0)
+
```

Comment:
I have some questions on this code block:

1. Why do we capture the container output only for the first phase?
2. Why do we use `print_progress()`, given that this output is not part of any conversion step?
3. If this log ingestion happens only in dev mode, why do we truncate it?
4. Why do we have a slightly different logging method in Qubes (`self.sanitize_conversion_str()` vs `util.replace_control_chars()`)?